docs/syscall.md

```diff
@@ -0,0 +1,17 @@
+# System Calls
+
+## SDL-Oriented System Calls
+
+### `syscall_draw_frame(screen, width, height)`
+Update the framebuffer of the given size, the window is lazy-initialized, if it does not exist when this system call is called, a new window will be created with the given width and height. This system call also poll events from SDL library, and update the internal input specific event queue if neccessary.
+
+### `syscall_draw_frame_pal(screen, pal, width, height)`
+Same as `syscall_draw_frame`, but with indexed-coloring, `pal` is a RGB look up table which contains 256 distinct colors, all pixels in the framebuffer will be automatically mapped before drawing on the screen.
+
+### `syscall_poll_event(event)`
+Poll a input-specific event from the event system, `event` should contains a 32-bit `type` field, and associated with a suitably sized value buffer, the internal event queue will be updated whenever `syscall_draw_frame` or `syscall_draw_frame_pal` is called.
+
+The following types of event are currently supported:
+* `KEY_EVENT`(`0x0`): Triggered when the status of keyboard is changed, either a key is pressed or released, returns a 32-bit keycode and a 8-bit key state, the hexadecimal keycode values are listed here: https://wiki.libsdl.org/SDLKeycodeLookup
```

Comment:
Replace "here: ..." with `in [SDL Keycode Lookup Table](https://wiki.libsdl.org/SDLKeycodeLookup)`.